### PR TITLE
Chore: Run betterer and eslint/prettier in parallel for pre-commit hooks

### PR DESCRIPTION
**Pull request by academo**
**What is this feature?**

Forces lint-staged to run betterer and eslint/prettier in parallel. This greatly reduces the pre-commit hook time when modifying javascript and typescript files

Lint staged supports task concurrency by [default](https://github.com/okonet/lint-staged#task-concurrency) warning about the risk of running commands that modify files at the same time. The eslint and prettier both modify the JS and TS files they inspect but betterer doesn't and is not concerned about the file style for its checks.

Betterer internally runs Eslint. This means that on each pre-commit eslint is running twice on the same files (though with a slight variation in rules). Because of the amount of plugins and rules we have in our eslint configuration, eslint bootstrap can take 2 of more seconds. By running in parallel betterer and eslint developers with multi-core computers (which are the norm) benefit greatly.

Since lint-staged requires the glob pattern to be the same as the configuration key. Because betterer only runs on ts and tsx files I created a new key to only lint-stage on those files, leaving js, ts and tsx for lint fixes (that must run in sequence)

#### The following tests were run with a single typescript file staged in `public/app`:

Benchmark of lint staged with sequencial betterer, eslint and prettier:

```
❯ hyperfine "yarn run lint-staged" --runs 10
Benchmark 1: yarn run lint-staged
  Time (mean ± σ):      6.989 s ±  0.114 s    [User: 16.687 s, System: 2.188 s]
  Range (min … max):    6.872 s …  7.166 s    10 runs
```

Benchmark of lint staged with parallel betterer, and sequencial eslint and prettier:

```
❯ hyperfine "yarn run lint-staged" --runs 10
Benchmark 1: yarn run lint-staged
  Time (mean ± σ):      5.284 s ±  0.036 s    [User: 17.303 s, System: 2.303 s]
  Range (min … max):    5.240 s …  5.370 s    10 runs
```

#### The following tests were run with a typescript file staged in `public/app` and a second one in `@grafana/data`

Benchmark of lint staged with sequencial betterer, eslint and prettier:

```
❯ hyperfine "yarn run lint-staged" --runs 10
Benchmark 1: yarn run lint-staged
  Time (mean ± σ):      7.020 s ±  0.064 s    [User: 16.617 s, System: 2.199 s]
  Range (min … max):    6.969 s …  7.187 s    10 runs
```

Benchmark of lint staged with parallel betterer, and sequencial eslint and prettier:

```
❯ hyperfine "yarn run lint-staged" --runs 10
Benchmark 1: yarn run lint-staged
  Time (mean ± σ):      5.515 s ±  0.085 s    [User: 17.984 s, System: 2.511 s]
  Range (min … max):    5.367 s …  5.621 s    10 runs
```

#### Improvement

Based on the benchmark this change reduces the pre-commit time between 1.5 and 1.8 seconds in a 12th gen intel machine (20% to 25% faster)

**Why do we need this feature?**

The existing setup takes tooo much time and it runs on every commit. It forces developers to wait additional seconds that are not necessary.

**Who is this feature for?**

All developers working in grafana javascript and typescript code

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer:**

Please check that:

- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: package.json
Comment (ashharrison90) [inline package.json:61]: this fake betterer extension makes me a little concerned tbh 😅 what happens if it's removed?

**Review by ashharrison90 — COMMENTED** [review]
1 question

File: package.json
Comment (academo) [reply package.json:61]: if you remove it then you will write an invalid JSON (two keys the same)

Comment (jackw) [reply package.json:61]: Looking at the globs in betterer.ts could we remove the `js` and `betterer` extensions here instead?

```
export default {
  'better eslint': () =>
    countEslintErrors()
      .include('**/*.{ts,tsx}')
      .exclude(/public\/app\/angular/),
  'no undocumented stories': () => countUndocumentedStories().include('**/!(*.internal).story.tsx'),
};
```

**Review by jackw — APPROVED** [review]
Took it for a spin and it shaved 5 seconds off `lint-staged` for me (intel i5 10th gen here!). I don't really see any harm in getting in this if it speeds up commits but would recommend waiting for others that know betterer better to review before merging.

**Without change**

```
hyperfine "yarn run lint-staged" --runs 10
Benchmark 1: yarn run lint-staged
  Time (mean ± σ):     14.571 s ±  0.591 s    [User: 21.014 s, System: 3.646 s]
  Range (min … max):   14.261 s … 16.229 s    10 runs
```
**With change**

```
hyperfine "yarn run lint-staged" --runs 10
Benchmark 1: yarn run lint-staged
  Time (mean ± σ):     10.815 s ±  0.212 s    [User: 22.451 s, System: 3.805 s]
  Range (min … max):   10.646 s … 11.384 s    10 runs
```

File: package.json
Comment (academo) [reply package.json:61]: This makes much more sense. I'll do that better.

**Review by ashharrison90 — APPROVED** [review]
feel better(er) now the magic file extension is gone 😄

**Pull request merged**
